runtime/util/ConfigurationParser.cpp

```diff
@@ -33,6 +33,11 @@ uint32_t getTagForSymbolName(const char *name) {
     return lb->second;
   }
   uint32_t const tag = getTagForSymbolNameInternal(s.c_str());
+  if (tag == ERROR_TAG) {
+    std::cerr << "No tag found for symbol " << name << ". Maybe attempted to "
+              << "evaluate a symbol with no rules?\n";
+    abort();
+  }
```

Comment:
How many places is this function called from? It might be better to delegate the reporting to _callers_ of this function; that way we can be more precise in the error message we print.

If there are lots of call sites and it would be a big change to do that, then I'm happy with this.

Reply:
It is called in two places in the configuration parser, in the functions that construct the initial configuration from text and binary KORE. It is also called quite some times in various runtime libraries (json, bytes, io, arithmetic, ffi). In all of these callsites, the symbol that is passed as an argument is a hardcoded symbol like `dotk{}`, some version of `inj{{.., ..}`, etc.. As such, I don't see much utility from specialized error messages.

Reply:
Great, this works absolutely fine for me in that case.